### DISCUSSION TITLE
An error of add ZHA on Homa Assistant create by ubuntu 22

### DISCUSSION THREAD

**Discussion by hwzolin** [zigpy/zha-device-handlers#1914] [Q&A]
Hi everyone, I have a question.
I created a Home Assistant with Python 3.10 on top of Ubuntu 22.04.1. When the dependency is wrapped and the home assistant is opened to add ZHA, two options will appear. As shown in the following figure:
![options](https://user-images.githubusercontent.com/42085859/201240173-37cc425b-7d35-4937-9353-c15ac438a94b.png)
If I click on the option above, the question "gpiozero.exc.BadPinFactory: Unable to load any default pin factory!" appears in the terminal, "gpiozero.exc.BadPinFactory: Unable to load any default pin factory!"
![failed to connect to localhost](https://user-images.githubusercontent.com/42085859/201241009-cbecfd43-b602-4a94-9063-7c9cb7d61e38.png)
And I noticed that there was also a prompt "PinFactoryFallback: Falling back from rpio: This module can only be run on a Raspberry Pi!"

If I select the option below, I will be prompted "Failed to connect" after the operation. But there is no information in the terminal.
![Failed to connect ZHA](https://user-images.githubusercontent.com/42085859/201241743-0a5243e1-c886-43b8-b027-809c46e74e5f.png)
I don't know what the problem is, I hope you can help based on it, thank you!

****
**Note: Below is all the information of the runtime terminal, please check it, thank you**

/usr/lib/python3/dist-packages/pkg_resources/__init__.py:116: PkgResourcesDeprecationWarning: 1.1build1 is an invalid version and will not be supported in a future release
  warnings.warn(
/usr/lib/python3/dist-packages/pkg_resources/__init__.py:116: PkgResourcesDeprecationWarning: 0.1.43ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
/usr/lib/python3/dist-packages/pkg_resources/__init__.py:116: PkgResourcesDeprecationWarning: 1.1build1 is an invalid version and will not be supported in a future release
  warnings.warn(
/usr/lib/python3/dist-packages/pkg_resources/__init__.py:116: PkgResourcesDeprecationWarning: 0.1.43ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
/usr/lib/python3/dist-packages/pkg_resources/__init__.py:116: PkgResourcesDeprecationWarning: 1.1build1 is an invalid version and will not be supported in a future release
  warnings.warn(
/usr/lib/python3/dist-packages/pkg_resources/__init__.py:116: PkgResourcesDeprecationWarning: 0.1.43ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
2022-11-11 09:26:08.310 WARNING (MainThread) [homeassistant.setup] Setup of bluetooth is taking over 10 seconds.
/usr/lib/python3/dist-packages/pkg_resources/__init__.py:116: PkgResourcesDeprecationWarning: 1.1build1 is an invalid version and will not be supported in a future release
  warnings.warn(
/usr/lib/python3/dist-packages/pkg_resources/__init__.py:116: PkgResourcesDeprecationWarning: 0.1.43ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
2022-11-11 09:26:24.566 WARNING (MainThread) [homeassistant.config_entries] Config entry 'MiWiFi router' for upnp integration not ready yet: Device not discovered: uuid:facf02ab-5afd-4af7-9820-98730a96475c::urn:schemas-upnp-org:device:InternetGatewayDevice:1; Retrying in background
/usr/lib/python3/dist-packages/pkg_resources/__init__.py:116: PkgResourcesDeprecationWarning: 1.1build1 is an invalid version and will not be supported in a future release
  warnings.warn(
/usr/lib/python3/dist-packages/pkg_resources/__init__.py:116: PkgResourcesDeprecationWarning: 0.1.43ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
/home/o/.homeassistant/deps/lib/python3.10/site-packages/gpiozero/devices.py:288: PinFactoryFallback: Falling back from rpigpio: This module can only be run on a Raspberry Pi!
  warnings.warn(
/home/o/.homeassistant/deps/lib/python3.10/site-packages/gpiozero/devices.py:288: PinFactoryFallback: Falling back from lgpio: module 'lgpio' has no attribute 'BOTH_EDGES'
  warnings.warn(
/home/o/.homeassistant/deps/lib/python3.10/site-packages/gpiozero/devices.py:288: PinFactoryFallback: Falling back from rpio: This module can only be run on a Raspberry Pi!
  warnings.warn(
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Can't connect to pigpio at localhost(8888)

Did you start the pigpio daemon? E.g. sudo pigpiod

Did you specify the correct Pi host/port in the environment
variables PIGPIO_ADDR/PIGPIO_PORT?
E.g. export PIGPIO_ADDR=soft, export PIGPIO_PORT=8888

Did you specify the correct Pi host/port in the
pigpio.pi() function? E.g. pigpio.pi('soft', 8888)
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
/home/o/.homeassistant/deps/lib/python3.10/site-packages/gpiozero/devices.py:288: PinFactoryFallback: Falling back from pigpio: failed to connect to localhost:8888
  warnings.warn(
/home/o/.homeassistant/deps/lib/python3.10/site-packages/gpiozero/devices.py:288: PinFactoryFallback: Falling back from native: unable to locate Pi revision in /proc/device-tree or /proc/cpuinfo
  warnings.warn(
2022-11-11 09:26:37.952 ERROR (MainThread) [homeassistant] Error doing job: Future exception was never retrieved
Traceback (most recent call last):
  File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/o/.homeassistant/deps/lib/python3.10/site-packages/zigpy_zigate/common.py", line 84, in set_pizigate_running_mode
    gpio0 = UnclosableOutputDevice(pin=GPIO_PIN0, initial_value=None)
  File "/home/o/.homeassistant/deps/lib/python3.10/site-packages/gpiozero/devices.py", line 108, in __call__
    self = super(GPIOMeta, cls).__call__(*args, **kwargs)
  File "/home/o/.homeassistant/deps/lib/python3.10/site-packages/zigpy_zigate/common.py", line 27, in __init__
    super().__init__(
  File "/home/o/.homeassistant/deps/lib/python3.10/site-packages/gpiozero/output_devices.py", line 83, in __init__
    super(OutputDevice, self).__init__(pin, pin_factory=pin_factory)
  File "/home/o/.homeassistant/deps/lib/python3.10/site-packages/gpiozero/mixins.py", line 85, in __init__
    super(SourceMixin, self).__init__(*args, **kwargs)
  File "/home/o/.homeassistant/deps/lib/python3.10/site-packages/gpiozero/devices.py", line 540, in __init__
    super(GPIODevice, self).__init__(**kwargs)
  File "/home/o/.homeassistant/deps/lib/python3.10/site-packages/gpiozero/devices.py", line 250, in __init__
    Device.pin_factory = Device._default_pin_factory()
  File "/home/o/.homeassistant/deps/lib/python3.10/site-packages/gpiozero/devices.py", line 291, in _default_pin_factory
    raise BadPinFactory('Unable to load any default pin factory!')
gpiozero.exc.BadPinFactory: Unable to load any default pin factory!

**Comment by Hedda** [comment]
Suggest trying posting to the Home Assistant community forum instead -> https://community.home-assistant.io/c/configuration/zigbee/39

Note that more detailed information about your specific will be needed there (you do not even mention Zigbee HW, computer, or OS setup).
